### DISCUSSION TITLE
Access mock S3 bucket using Virtual Hosted-Style

### DISCUSSION THREAD

**Discussion by Ken-Miura** [awslabs/aws-sdk-rust#485] [Q&A] [answered]
I set up mock S3 bucket using MinIO in local docker environment, and configured setting so that it can handle Virtual Hosted-Style access (I set up referring to Domain section in https://docs.min.io/docs/minio-server-configuration-guide.html). But I don't understand how to access to the mock bucket using this sdk.

I checked this post (https://github.com/awslabs/aws-sdk-rust/discussions/253#discussioncomment-1478233), But It looks like this post is assumed to use Path-Style Requests (After changing endpoint, I got packet capture, and then checked Path-Style Request is used). I heard Path-Style will be deprecated (https://aws.amazon.com/jp/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/). So I would like to use Virtual Hosted-Style in my app. How can I use Virtual Hosted-Style for access to local mock S3?

**Comment by Velfi** [comment]
The SDK doesn't currently support virtual addressing but it's on our roadmap.

**Reply by Ken-Miura** [reply]
Thanks. I look forward to Virtual Hosted-Style support.

**Reply by Ken-Miura** [reply, marked as answer]
It seems that Virtual Hosted-Style is now supported from below release according to release note.
https://github.com/awslabs/aws-sdk-rust/releases/tag/release-2023-01-13
